deeptime/decomposition/_koopman.py

```diff
@@ -670,41 +675,105 @@ def feature_component_correlation(self):
+            Limit the number of default observables (and of default statistics) to this number.
+            Only used if `observables` are None or `statistics` are None.
+        observables : (input_dimension, n_observables) ndarray
+            Coefficients that express one or multiple observables in
+            the basis of the input features.
+        statistics : (input_dimension, n_statistics) ndarray
+            Coefficients that express one or multiple statistics in the basis of the input features.
+        progress : ProgressBar, optional, default=None
+            Optional progress bar, tested for tqdm.
 
-    """
+        Returns
+        -------
+        test : deeptime.util.validation.ChapmanKolmogorovTest
+            The test results
+
+        See Also
+        --------
+        :meth:`ck_test <deeptime.util.validation.ck_test>`
+
+        Notes
+        -----
+        This method computes two sets of time-lagged covariance matrices
+
+        * estimates at higher lag times :
+
+          .. math::
+
+              \left\langle \mathbf{K}(n\tau)g_{i},f_{j}\right\rangle_{\rho_{0}}
+
+          where :math:`\rho_{0}` is the empirical distribution implicitly defined
+          by all data points from time steps 0 to T-tau in all trajectories,
+          :math:`\mathbf{K}(n\tau)` is a rank-reduced Koopman matrix estimated
+          at the lag-time n*tau and g and f are some functions of the data.
 
-    def __init__(self, test_model: Model, test_estimator: Estimator, test_model_lagtime: int, mlags,
-                 observables, statistics, observables_mean_free, statistics_mean_free):
-        super().__init__(test_model, test_estimator, test_model_lagtime, mlags)
+        * predictions at higher lag times :
+
+          .. math::
+
+              \left\langle \mathbf{K}^{n}(\tau)g_{i},f_{j}\right\rangle_{\rho_{0}}
+
+          where :math:`\mathbf{K}^{n}` is the n'th power of the rank-reduced
+          Koopman matrix contained in self.
+
+        The Champan-Kolmogorov test is to compare the predictions to the estimates.
+        """
+        test_model = self if test_model is None else test_model
+        if n_observables is not None:
+            if n_observables > test_model.dim:
+                import warnings
+                warnings.warn('Selected singular functions as observables but dimension '
+                              'is lower than requested number of observables.')
+                n_observables = test_model.dim
+        else:
+            n_observables = test_model.dim
+
+        if isinstance(observables, str) and observables == 'phi':
+            observables = test_model.singular_vectors_right[:, :n_observables]
+            observables_mean_free = True
+        else:
+            observables_mean_free = False
+
+        if isinstance(statistics, str) and statistics == 'psi':
+            statistics = test_model.singular_vectors_left[:, :n_observables]
```

Comment:
Isn't one of these arrays (left/right singular vectors) transposed?

Reply:
i think it's correct like this but perhaps i should add a test at some point checking that...